ai21/clients/vertex/gcp_authorization.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+from typing import Optional
+
+import google.auth
+from google.auth.credentials import Credentials
+from google.auth.transport.requests import Request
+
+
+class GCPAuthorization:
+    def __init__(self, credentials: Optional[Credentials] = None, project_id: Optional[str] = None):
+        self._credentials = credentials
+        self._project_id = project_id
+
+    def _set_auth(self):
+        credentials, loaded_project_id = google.auth.default(
+            scopes=["https://www.googleapis.com/auth/cloud-platform"],
+        )
+
+        if self._project_id is not None and self._project_id != loaded_project_id:
```

Comment:
`self._project_id is not None` is not necessary no? Can `loaded_project_id` be `None`?

Reply:
`loaded_project_id` can be None (when it's none, the google sdk only logs a warning)